commit 3926ef0b4dca07b0bafed2714710bb39f2fc5bd2
Author: q66 <q66@chimera-linux.org>
Date:   Sat Feb 14 10:36:45 2026 +0100

    revert broken tests change
    
    Reverts https://github.com/qt/qtbase/commit/f859bd4a217a124f4c41167779036c6eb69ca49b
    
    This stuff doesn't even build. The relocatable_change.h fails
    due to <QtTest/QTest> not being found. I dunno how this was tested
    but just revert it.

diff --git a/tests/auto/corelib/kernel/qvariant/CMakeLists.txt b/tests/auto/corelib/kernel/qvariant/CMakeLists.txt
index d79c5d3..eae9c0d 100644
--- a/tests/auto/corelib/kernel/qvariant/CMakeLists.txt
+++ b/tests/auto/corelib/kernel/qvariant/CMakeLists.txt
@@ -42,9 +42,3 @@ qt_internal_extend_target(tst_qvariant CONDITION NOT QT_FEATURE_doubleconversion
     DEFINES
         QT_NO_DOUBLECONVERSION
 )
-
-if(BUILD_SHARED_LIBS AND QT_FEATURE_library)
-    qt_add_library(tst_qvariant_relocatabilitychange MODULE relocatable_change.cpp)
-    target_link_libraries(tst_qvariant_relocatabilitychange PRIVATE Qt6::Core)
-    add_dependencies(tst_qvariant tst_qvariant_relocatabilitychange)
-endif()
diff --git a/tests/auto/corelib/kernel/qvariant/tst_qvariant.cpp b/tests/auto/corelib/kernel/qvariant/tst_qvariant.cpp
index 3455305..6736f96 100644
--- a/tests/auto/corelib/kernel/qvariant/tst_qvariant.cpp
+++ b/tests/auto/corelib/kernel/qvariant/tst_qvariant.cpp
@@ -82,10 +82,6 @@ CHECK_GET(MyVariant, const &&);
 #include <QUrl>
 #include <QUuid>
 
-#if QT_CONFIG(library) && defined(QT_SHARED)
-#  include <QLibrary>
-#endif
-
 #include <private/qcomparisontesthelper_p.h>
 #include <private/qlocale_p.h>
 #include <private/qmetatype_p.h>
@@ -97,9 +93,6 @@ CHECK_GET(MyVariant, const &&);
 #include <variant>
 #include <unordered_map>
 
-#define WHICH_TYPE_IS_RELOCATABLE RelocatableInAppType
-#include "relocatable_change.h"
-
 using namespace Qt::StringLiterals;
 
 class CustomNonQObject;
@@ -387,9 +380,6 @@ private slots:
     void saveInvalid();
     void saveNewBuiltinWithOldStream();
 
-    void relocatabilityChange_data();
-    void relocatabilityChange();
-
     void implicitConstruction();
 
     void iterateSequentialContainerElements_data();
@@ -4070,7 +4060,7 @@ struct MyNotMovable
         return ok;
     }
     // Make it too big to store it in the variant itself
-    char dummy[QVariant::Private::MaxInternalSize * 2];
+    void *dummy[4];
 };
 
 int MyNotMovable::count  = 0;
@@ -4081,7 +4071,6 @@ struct MyShared : QSharedData {
 
 QT_BEGIN_NAMESPACE
 Q_DECLARE_TYPEINFO(MyMovable, Q_RELOCATABLE_TYPE);
-Q_DECLARE_TYPEINFO(RelocatableInAppType, Q_RELOCATABLE_TYPE);
 QT_END_NAMESPACE
 
 Q_DECLARE_METATYPE(MyPrimitive)
@@ -4704,6 +4693,7 @@ void tst_QVariant::saveNewBuiltinWithOldStream()
     QCOMPARE(int(data.constData()[3]), 0);
 }
 
+#if 0
 using PluginCreateVariantFn = QVariant (*)(bool relocatable);
 static PluginCreateVariantFn pluginCreateVariant = nullptr;
 void tst_QVariant::relocatabilityChange_data()
@@ -4804,6 +4794,7 @@ void tst_QVariant::relocatabilityChange()
         QCOMPARE_EQ(copy.constData(), variant.constData());
     }
 }
+#endif
 
 template<typename Container, typename Value_Type = typename Container::value_type>
 struct ContainerAPI
